### PR TITLE
Cleaned up the Makefile and removed the .dockerignore

### PR DESCRIPTION
**Pull request by carnarez**

### CODE REVIEW

File: py-polars/Makefile [hunk 1]
```diff
@@ -1,10 +1,28 @@
...
+release-install: release-build pip
+
+test:
+	maturin develop
+	pytest tests
+
+test-build:
+	maturin build -o wheels -i $(which python)
+
+test-install: test-build pip
```
Comment (ritchie46) [inline py-polars/Makefile:25]: I think we can also remove some unused tasks: this one

File: py-polars/Makefile [hunk 2]
```diff
@@ -1,10 +1,28 @@
...
+
+build-and-test: test-install
```
Comment (ritchie46) [inline py-polars/Makefile:27]: and this one. It is redundant with `make test`

**Comment by ritchie46** [comment]
LGTM. :+1:  I have a few small nits and the CI/CD is failing because it relied on the old `tasks.sh` file, so we should use the new make task for that.

File: py-polars/Makefile (hunk 1 above)
Comment (carnarez) [reply py-polars/Makefile]: Done

File: py-polars/Makefile (hunk 2 above)
Comment (carnarez) [reply py-polars/Makefile]: Done

**Comment by carnarez** [comment]
> LGTM. 👍 I have a few small nits and the CI/CD is failing because it relied on the old `tasks.sh` file, so we should use the new make task for that.

Fixed in the new commit.

File: py-polars/Makefile (hunk 1 above)
Comment (carnarez) [reply py-polars/Makefile:25]: Rolled back in.

File: py-polars/Makefile (hunk 2 above)
Comment (carnarez) [reply py-polars/Makefile]: Rolled back in.

**Pull request merged**